vim-canvas-demo-app-react/src/components/settings-app/BrandCustomization.tsx

```diff
@@ -9,9 +9,21 @@ export const BrandCustomization: React.FC = () => {
   const organizationId = useOrganizationContext();
 
   const handleSave = useCallback(() => {
-    saveSettings({ [organizationId]: { appColor } });
+    saveSettings({
+      organization_id: organizationId,
+      theme_color: appColor,
+    });
   }, [organizationId, appColor]);
 
+  useEffect(() => {
+    const fetchSettings = async () => {
+      const settings = await loadSettings(organizationId);
+      setAppColor((prev) =>
+        settings?.theme_color ? settings.theme_color : prev
```

Comment:
What will happen if the user will change the color before the settings are loaded?

Reply:
How will they be the same? If the user selects something other than the saved color?